lib/logic/player/content.dart

```diff
@@ -381,54 +381,62 @@ class ContentControl extends Control {
-        if (origin is Playlist && state.playlists.firstWhereOrNull((el) => el == origin) == null) {
-          QueueControl.instance.resetQueueAsFallback();
-        }
-        sort(emitChangeEvent: false, contentType: contentType);
-        break;
-      case ContentType.artist:
-        state.artists = (await SweyerPlugin.instance.retrieveArtists(Artist.fromMap)).toList();
-        if (disposed.value) {
-          return;
-        }
-        final origin = QueueControl.instance.state.origin;
-        if (origin is Artist && state.artists.firstWhereOrNull((el) => el == origin) == null) {
-          QueueControl.instance.resetQueueAsFallback();
-        }
-        sort(emitChangeEvent: false, contentType: contentType);
-        break;
-    }
-    if (emitChangeEvent) {
-      emitContentChange();
+    try {
+      switch (contentType) {
+        case ContentType.song:
+          state.allSongs.setSongs((await SweyerPlugin.instance.retrieveSongs(Song.fromMap)).toList());
+          if (_empty) {
+            dispose();
+            return;
+          }
+          sort(emitChangeEvent: false, contentType: contentType);
+          if (updateQueues) {
+            QueueControl.instance.removeObsolete(emitChangeEvent: false);
+          }
+          break;
+        case ContentType.album:
+          state.albums = await SweyerPlugin.instance.retrieveAlbums(Album.fromMap);
+          if (disposed.value) {
+            return;
+          }
+          final origin = QueueControl.instance.state.origin;
+          if (origin is Album && state.albums[origin.id] == null) {
+            QueueControl.instance.resetQueueAsFallback();
+          }
+          sort(emitChangeEvent: false, contentType: contentType);
+          break;
+        case ContentType.playlist:
+          state.playlists = (await SweyerPlugin.instance.retrievePlaylists(Playlist.fromMap)).toList();
+          if (disposed.value) {
+            return;
+          }
+          final origin = QueueControl.instance.state.origin;
+          if (origin is Playlist && state.playlists.firstWhereOrNull((el) => el == origin) == null) {
+            QueueControl.instance.resetQueueAsFallback();
+          }
+          sort(emitChangeEvent: false, contentType: contentType);
+          break;
+        case ContentType.artist:
+          state.artists = (await SweyerPlugin.instance.retrieveArtists(Artist.fromMap)).toList();
+          if (disposed.value) {
+            return;
+          }
+          final origin = QueueControl.instance.state.origin;
+          if (origin is Artist && state.artists.firstWhereOrNull((el) => el == origin) == null) {
+            QueueControl.instance.resetQueueAsFallback();
+          }
+          sort(emitChangeEvent: false, contentType: contentType);
+          break;
+      }
+      if (emitChangeEvent) {
+        emitContentChange();
+      }
+    } on SweyerPluginException catch (ex, stack) {
+      FirebaseCrashlytics.instance.recordError(
```

Comment:
I don't think this method should catch exceptions.
Exception here means something has gone incredibly wrong.

What will user see if on start up this has thrown an error?

Also, this method is called from more places than just `init`. It's not a responsibility of this method to decide what to do with the error.

Reply:
Actually, I think it **can** catch the error, but it should then show a toast to a user that something went wrong.

Reply:
Yeah, I think we must catch the error here or check every call site and make sure to catch the error there. Because during the conversion from Java to Android I did have some instances where the native side threw an error and the App would just be stuck in the launch screen, which is not a good user experience.

But I do agree that we need to show something if possible, so let's add a toast.